kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java

```diff
@@ -218,7 +246,9 @@ public void onClosed(WebSocket webSocket, int code, String reason) {
         webSocketClosed.set(true);
         LOGGER.debug("Exec Web Socket: On Close with code:[{}], due to: [{}]", code, reason);
         try {
-            closeExecutor();
+            if (cleanupOnClose.get()) {
```

Comment:
If we close the streams before the pumper, the pumper will consider this an error and will propagate Exception messages like `Pipe closed`, so we have to always close the pumper first and then the streams.

Now, if we implicitly close the streams, the user code may be still reading and receive a `Pipe closed` message.

I think that since we are handing the user a stream and a handle, we should not close the stream until the user decides to.